fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/LogicalTypeFilterExpression.java

```diff
@@ -134,7 +161,47 @@ public Iterable<MatchInfo> subsumedBy(@Nonnull final RelationalExpression candid
                                           @Nonnull final AliasMap aliasMap,
                                           @Nonnull final IdentityBiMap<Quantifier, PartialMatch> partialMatchMap,
                                           @Nonnull final EvaluationContext evaluationContext) {
-        return exactlySubsumedBy(candidateExpression, aliasMap, partialMatchMap);
+        final Iterable<MatchInfo> subsumedMatches = exactlySubsumedBy(candidateExpression, aliasMap, partialMatchMap);
+        if (!(candidateExpression instanceof LogicalTypeFilterExpression)) {
+            return subsumedMatches;
+        }
+        LogicalTypeFilterExpression typeFilterExpression = (LogicalTypeFilterExpression) candidateExpression;
+        if (typeFilterExpression.predicates.stream().noneMatch(predicate -> predicate instanceof Placeholder) || recordTypes.size() != 1) {
+            return subsumedMatches;
+        }
+
+        // Bind any predicate that can be bound to the record type key
+        final String recordTypeName = Iterables.getOnlyElement(recordTypes);
+        Comparisons.Comparison comparison = new RecordTypeKeyComparison.RecordTypeComparison(recordTypeName);
+        Optional<ComparisonRange> comparisonRangeMaybe = ComparisonRange.from(comparison);
+        if (comparisonRangeMaybe.isEmpty()) {
+            return subsumedMatches;
+        }
+        ComparisonRange comparisonRange = comparisonRangeMaybe.get();
+        QueryPredicate predicate = new ValuePredicate(new RecordTypeValue(inner.getAlias()), comparison);
```

Comment:
Shouldn't this predicate already be in your set of query predicates that are part of the query side? You can create it when the type filter expression with the appropriate `recordTypes` is created.